src/arrayMethodReduce.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, ...initialValue) {
+    let begin = 0;
+    let returnedValue = initialValue[0];
+
+    if (initialValue[0] === undefined) {
+      returnedValue = 'undefined';
+    }
+
+    if (initialValue.length === 0) {
+      returnedValue = this[0];
+      begin++;
+    }
```

Comment:
Як на мене ці дві перевірки можна об'єднати в одну

```suggestion
     if (arguments.length === 1) {
      begin = 1;
      returnedValue = this[0];
    }
```